### PR TITLE
nautilus: os/bluestore: go beyond pinned onodes while trimming the cache.

### PR DESCRIPTION
**Pull request by ifed01**
One might face lack of cache trimming when there is a bunch of pinned entries on the top of Onode's cache LRU list. If these pinned entries stay in the state for a long time cache might start using too much memory causing OSD to go out of osd-memory-target limit. Pinned state tend to happen to osdmap onodes.
The proposed patch preserves last trim position in the LRU list (if it pointed to a pinned entry) and proceeds trimming from that position if it wasn't invalidated. LRU nature of the list enables to do that safely since no new entries appear above the previously present entry while it's not touched.

This patch is specific to Nautilus since Octopus+ releases use a different pinning/cache trimming approach.

Fixes: https://tracker.ceph.com/issues/48729
Signed-off-by: Igor Fedotov <ifedotov@suse.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/common/options.cc [hunk 1]
```diff
@@ -4914,7 +4914,9 @@ std::vector<Option> get_global_options() {
     Option("bluestore_debug_small_allocations", Option::TYPE_INT, Option::LEVEL_DEV)
     .set_default(0)
     .set_description(""),
-
+    Option("bluestore_debug_max_cached_onodes", Option::TYPE_INT, Option::LEVEL_DEV)
+    .set_default(0)
+    .set_description(""),
```
Comment (aclamk) [inline src/common/options.cc:4919]: We need an explanation what it does.

**Review by aclamk — CHANGES_REQUESTED** [review]
Good patching stuff.

**Review by aclamk — APPROVED** [review]
I like this solution. 
I guess it is best we can do without backporting huge parts of logic.

**Comment by yuriw** [comment]
[wip-yuri7-testing-2021-03-02-1118-nautilus](https://trello.com/c/0lcwQSBD/1164-wip-yuri7-testing-2021-03-02-1118-nautilus)

**Pull request merged**